hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java

```diff
@@ -171,6 +178,8 @@ public Response findAlerts(
             final UriInfo uri) {
         Pager pager = RequestUtil.extractPaging(uri);
         try {
+            ResponseUtil.checkForUnknownQueryParams(uri, PARAMS_FIND_ALERTS);
```

Comment:
If there is a way to drive off of the annotations that is fine by me.  I agree that this is little bit brute force.  On the plus side it is explicit and can be applied as desired to certain endpoints.  It is not unusual to ignore unknown query params in REST API's, so it may be that we only want to apply this sort of check in specific places, like when using criteria for fetch or delete.